dallinger/mturk.py

```diff
@@ -182,104 +248,112 @@ def get_qualification_type_by_name(self, name):
+        self.update_qualification_score(qtype_id, worker_id, new_score)
 
         return {
             'qtype': result['qtype'],
             'score': new_score
         }
 
-    def update_qualification_score(self, qualification_id, worker_id, score):
-        """Score a worker for a specific qualification"""
-        return self._is_ok(self.mturk.update_qualification_score(
-            qualification_id,
-            worker_id,
-            score,
+    def revoke_qualification(self, qualification_id, worker_id, reason=''):
+        return self._is_ok(self.mturk.disassociate_qualification_from_worker(
+            QualificationTypeId=qualification_id,
+            WorkerId=worker_id,
+            Reason=reason
         ))
 
+    def get_qualification_score(self, qualification_id, worker_id):
+        """Return a worker's qualification score as an iteger.
+        """
+        try:
+            response = self.mturk.get_qualification_score(
+                QualificationTypeId=qualification_id,
+                WorkerId=worker_id
+            )
+        except ClientError as ex:
+            raise MTurkServiceException(ex.message)
+        return response['Qualification']['IntegerValue']
+
+    def get_current_qualification_score(self, name, worker_id):
+        """Return the current score for a worker, on a qualification with the
+        provided name.
+        """
+        qtype = self.get_qualification_type_by_name(name)
+        if qtype is None:
+            raise QualificationNotFoundException(
+                'No Qualification exists with name "{}"'.format(name)
+            )
+        try:
+            score = self.get_qualification_score(qtype['id'], worker_id)
+        except MTurkServiceException:
+            score = None
+        return {
+            'qtype': qtype,
+            'score': score
+        }
+
     def dispose_qualification_type(self, qualification_id):
         """Remove a qualification type we created"""
-        return self._is_ok(
-            self.mturk.dispose_qualification_type(qualification_id)
+        return self.mturk.delete_qualification_type(
+            QualificationTypeId=qualification_id
         )
 
     def get_workers_with_qualification(self, qualification_id):
         """Get workers with the given qualification."""
         done = False
-        page = 1
+        next_token = None
         while not done:
-            res = self.mturk.get_qualifications_for_qualification_type(
-                qualification_id,
-                page_size=100,
-                page_number=page
-            )
-            if res:
-                for r in res:
-                    yield {'id': r.SubjectId, 'score': r.IntegerValue}
-                page = page + 1
+            if next_token is not None:
+                response = self.mturk.list_workers_with_qualification_type(
+                    QualificationTypeId=qualification_id,
+                    MaxResults=100,
```

Comment:
Is this the page size, or a new limit on the total number of results?

Reply:
Page size. I changed this to a constant.